packages/block-editor/src/store/selectors.js

```diff
@@ -1501,6 +1501,18 @@ export function getSettings( state ) {
 	return state.settings;
 }
 
+/**
+ * Returns the block editor feature by name.
+ *
+ * @param {Object} state       Editor state.
+ * @param {string} featureName Feature name.
+ *
+ * @return {Object} The block editor feature object.
+ */
+export function __experimentalGetFeature( state, featureName ) {
```

Comment:
How about the following:
```js
export function __experimentalGetFeature( state, featureName, blockName ) {
```

I didn't want to cover the per block approach in the first iteration because of the legacy considerations with the block editor setting that needs to be addressed first.

How about we land an initial version and then I can branch and open two PRs:
- sync story for existing settings
- drop cap implementation that works also on a block type level?

Reply:
Yeah, that works for me if we keep things experimental.

> export function __experimentalGetFeature( state, featureName, blockName )

Is the blockName the only thing that could affect the result? Maybe it is, just trying to think ahead.

Reply:
The one thing I'm not sure about is whether we always want to return an object with a feature's config.

I'd be interested to see the following as well:

```js
const isDropCapAllowed = wp.data.select( 'core/block-editor' ).__experimentalGetFeature( 'typography.allowDropCap, 'core/paragraph' );
```

Reply:
> How about we land an initial version and then I can branch and open two PRs:
> 
>     * sync story for existing settings
> 
>     * drop cap implementation that works also on a block type level?

Sounds good.